src/test/steps/createCampaignSteps.ts

```diff
@@ -0,0 +1,12 @@
+import { When } from '@cucumber/cucumber';
+import basePage from '../../pages/basePage';
+
+When('User selects {string} for {string} on {string}', async function (company, type, page) {
+    const [companyName, companyType, pageName] = basePage.wrapper.toCamelCase(company, type, page);
+    await basePage.wrapper.waitAndClick(basePage[pageName][companyType]) 
+
+    await basePage.wrapper.waitAndClick(basePage[pageName].companyNameInput) 
+    await basePage.wrapper.type(basePage[pageName].companyNameInput, company);
+    await basePage.wrapper.pressKeyboard('Enter');
+    await basePage.wrapper.waitAndClick(basePage[pageName][companyName]);
+});
```

Comment:
Does this step definition work in case you re-use it anywhere? For example:
```Gherkin
User selects "Business" for "Industry input" on "Sign up page"
```

Reply:
This step covers only this section:
![image](https://github.com/YahyoMahmudov/E2E-Testing/assets/109512781/1e89c94b-5b1f-4b04-bce6-ffda113e72ec)

Reply:
What if I want to create a campaign for another company in another scenario? Will it still work or the test fail?

Reply:
Yes, it will work for other companies also

Reply:
Pay attention to step definition parameters!!